src/raudio.c

```diff
@@ -431,8 +431,10 @@ static bool SaveFileText(const char *fileName, char *text);         // Save text
 AudioBuffer *LoadAudioBuffer(ma_format format, ma_uint32 channels, ma_uint32 sampleRate, ma_uint32 sizeInFrames, int usage);
 void UnloadAudioBuffer(AudioBuffer *buffer);
 
+bool IsAudioBufferPlayingInLockedState(AudioBuffer *buffer);
```

Comment:
Why are those functions required? Could they just be avoided?

Reply:
This is what I mean in the description with 
> This required separation of some of the functions, as they are called from both threads and **mutex is not re-entrant**.

Calling lock twice within a call-stack will end up in a deadlock.

Reply:
Ok, I see, in that case, could those functions be kept internal to raudio module?

Reply:
Those functions are "internal" - or what do you mean? Do you mean to make them `static` ?